acceptance/acceptance_test.go

```diff
@@ -229,1905 +99,1417 @@ func testWithoutSpecificBuilderRequirement(
+			assert.Contains(output, "invalid builder toml")
+		})
+	})
+
+	when("creating a windows builder", func() {
+		var tmpDir string
+
+		it.Before(func() {
+			h.SkipIf(t, containerManager.HostOS() != "windows", "The current Docker daemon does not support Windows-based containers")
+
+			var err error
+			tmpDir, err = ioutil.TempDir("", "create-test-builder")
+			assert.Nil(err)
+		})
+
+		it.After(func() {
+			os.RemoveAll(tmpDir)
+		})
+
+		when("experimental is disabled", func() {
+			it("fails", func() {
+				stackManager.EnsureDefaultStackCreated()
+
+				buildpack := components.NoOpBuildpack
+				buildpackManager.PlaceBuildpacksInDir(tmpDir, buildpack)
+				builderConfigPath := builderManager.NewDynamicBuilderConfig(buildpack).ConfigFile(tmpDir)
+
+				output, err := createBuilderPack.RunWithCombinedOutput(
+					"create-builder",
+					"--no-color",
+					registry.RepoName(fmt.Sprintf("test/builder-%s", h.RandString(8))),
+					"-b", builderConfigPath,
+				)
+
+				assert.NotNil(err)
+				assert.NewOutputAssertionManager(output).ReportsWindowsContainersExperimental()
+			})
+		})
+
+		when("experimental is enabled", func() {
+			it("succeeds", func() {
+				stackManager.EnsureDefaultStackCreated()
+
+				createBuilderPack.EnableExperimental()
+
+				testBuilder := builderManager.CreateWindowsBuilder()
+				defer testBuilder.Cleanup()
+
+				assert.ImageIsWindows(testBuilder.Name())
+			})
+		})
 	})
 
-	when("stack is created", func() {
+	when("build", func() {
 		var (
-			runImageMirror string
+			assertAppImage          assertions.ImageAssertionManager
+			appName, appImageName   string
+			mockAppPath             = filepath.Join("testdata", "mock_app")
+			mixedComponentMessenger components.MixedComponents
+			cacheCleanupRequired    bool
 		)
 
 		it.Before(func() {
-			value, err := suiteManager.RunTaskOnceString("create-stack",
-				func() (string, error) {
-					runImageMirror := registryConfig.RepoName(runImage)
-					err := createStack(t, dockerCli, runImageMirror)
-					if err != nil {
-						return "", err
-					}
-
-					return runImageMirror, nil
-				})
-			h.AssertNil(t, err)
+			cacheCleanupRequired = false
+			h.SkipIf(t, containerManager.HostOS() == "windows", "These tests are not yet compatible with Windows-based containers")
 
-			suiteManager.RegisterCleanUp("remove-stack-images", func() error {
-				return h.DockerRmi(dockerCli, runImage, buildImage, value)
-			})
+			appName = fmt.Sprintf("some-org/%s", h.RandString(10))
+			appImageName = registry.RepoName(appName)
+
+			assertAppImage = assert.NewImageAssertionManager(appImageName, containerManager)
+
+			mixedComponentMessenger = components.NewMixedComponents(t, testLifecycle, pack)
+
+			cacheCleanupRequired = true
+		})
+
+		it.After(func() {
+			if cacheCleanupRequired {
+				h.DockerRmi(dockerCli, appImageName)
+				ref, err := name.ParseReference(appImageName, name.WeakValidation)
+				assert.Nil(err)
+				cacheImage := cache.NewImageCache(ref, dockerCli)
```

Comment:
Maybe it's worth adding a comment explaining what we are doing here? It seems a little strange to be creating a new thing in the test cleanup, but I assume that is so we can call `Clear` on it.

Reply:
I agree this is a confusing pattern, but it's what we have today, so I'd opt to keep it out of scope for this PR.

Reply:
I thought that since you're trying to make the code clearer in this PR, it's ok to add a comment in this case, but I don't have a strong opinion on that.
If you decide not to add it, maybe it's better to create an issue and to specify unclear code that we want to explain.